alfresco-search/src/main/java/org/alfresco/solr/SolrInformationServer.java

```diff
@@ -4114,256 +2915,960 @@ public void reindexNodeByQuery(String query) throws IOException, AuthenticationE
+
+                    if (cachedDoc != null)
+                    {
+                        updatePathRelatedFields(nodeMetaData, cachedDoc);
+                        updateNamePathRelatedFields(nodeMetaData, cachedDoc);
+                        updateAncestorRelatedFields(nodeMetaData, cachedDoc);
+
+                        AddUpdateCommand addDocCmd = new AddUpdateCommand(request);
+                        addDocCmd.overwrite = overwrite;
+                        addDocCmd.solrDoc = cachedDoc;
+
+                        processor.processAdd(addDocCmd);
+                        solrContentStore.storeDocOnSolrContentStore(fixedTenantDomain, nodeMetaData.getId(), cachedDoc);
+                    }
+                    else
+                    {
+                        LOGGER.debug("No child doc found to update {}", childId);
+                    }
+                }
+                finally
+                {
+                    unlock(childId);
+                }
+            }
+        }
+    }
+
+
+    private long topNodeId(String sortDir)
+    {
+        try (SolrQueryRequest request = this.newSolrQueryRequest())
+        {
+            ModifiableSolrParams params =
+                    new ModifiableSolrParams(request.getParams())
+                            .set(CommonParams.Q, FIELD_DOC_TYPE + ":" + DOC_TYPE_NODE)
+                            .set(CommonParams.ROWS, 1)
+                            .set(CommonParams.SORT, FIELD_DBID + " " + sortDir)
+                            .set(CommonParams.FL, FIELD_DBID);
+
+            SolrDocumentList docs = cloud.getSolrDocumentList(nativeRequestHandler, request, params);
+            return docs.stream()
+                    .findFirst()
+                    .map(doc -> getFieldValueLong(doc, FIELD_DBID))
+                    .orElse(0L);
         }
+    }
 
-        public boolean needsScores() {
-            return false;
+    private void initSkippingDescendantDocs(Properties p, Set<QName> dataForSkippingDescendantDocs, String propPrefixParent,
+                                            String skipByField, DefinitionExistChecker dec)
+    {
+        int i = 0;
+        for (String key = new StringBuilder(propPrefixParent).append(i).toString(); p.containsKey(key);
+             key = new StringBuilder(propPrefixParent).append(++i).toString())
+        {
+            String qNameInString = p.getProperty(key);
+            if ((null != qNameInString) && !qNameInString.isEmpty())
+            {
+                QName qName = QName.resolveToQName(dataModel.getNamespaceDAO(), qNameInString);
+
+                LOGGER.warning("QName was not found for {}", qNameInString);
+
+                if (dec.isDefinitionExists(qName))
+                {
+                    dataForSkippingDescendantDocs.add(qName);
+                }
+            }
         }
 
-        public void collect(int doc) throws IOException
+        if (null == skippingDocsQueryString)
```

Comment:
Is skippingDocsQueryString a member variable?  Using a member variable in a private method in such a large class seems fairly unreadable.  Perhaps this method could return the string instead?